dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/datasetutility/ReplaceFileHandlerIT.java

```diff
@@ -53,6 +101,138 @@ public void shouldCreateDataFile() {
 
     }
 
+    @Test
+    public void shouldReplaceFile() throws IOException {
+        //given
+        dataverseSession.setUser(authenticationServiceBean.getAdminUser());
+
+        Dataverse dataverse = new Dataverse();
+        fillDataverseWithRequiredData(dataverse);
+        em.persist(dataverse);
+
+        Dataset dataset = new Dataset();
+        dataset.setOwner(dataverse);
+        fillDatasetWithExtendedData(dataset);
+        em.persist(dataset);
+        em.flush();
+
+        DataFile initialFile = createTestDataFile(dataset.getLatestVersion(), "banner", "image/png", true);
+        em.persist(initialFile);
+
+
+        byte[] bytes = IOUtils.resourceToByteArray("images/coffeeshop.png", getClass().getClassLoader());
+        File newfile = new File(FileUtil.getFilesTempDirectory() +"/coffeeshop.png");
+        try {
+            Files.write(newfile.toPath(), bytes);
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
```

Comment:
Do not ignore exception. There is no point in going further if this will fail. Just:
```
        byte[] bytes = IOUtils.resourceToByteArray("images/coffeeshop.png", getClass().getClassLoader());
        File newfile = new File(FileUtil.getFilesTempDirectory() +"/coffeeshop.png");
        Files.write(newfile.toPath(), bytes);
```
will be enough